modules/integration/tests-integration/tests-server/src/test/java/org/wso2/mb/integration/tests/server/mgt/MetricsTestCase.java

```diff
@@ -0,0 +1,458 @@
+import org.wso2.carbon.metrics.impl.MetricsLevelConfigException;
+import org.wso2.carbon.metrics.manager.jmx.MetricManagerMXBean;
+import org.wso2.carbon.metrics.view.ui.MetricDataWrapper;
+import org.wso2.carbon.metrics.view.ui.MetricsViewClient;
+import org.wso2.mb.integration.common.clients.AndesClient;
+import org.wso2.mb.integration.common.clients.configurations.AndesJMSConsumerClientConfiguration;
+import org.wso2.mb.integration.common.clients.configurations.AndesJMSPublisherClientConfiguration;
+import org.wso2.mb.integration.common.clients.exceptions.AndesClientConfigurationException;
+import org.wso2.mb.integration.common.clients.exceptions.AndesClientException;
+import org.wso2.mb.integration.common.clients.operations.utils.ExchangeType;
+import org.wso2.mb.integration.common.utils.backend.MBIntegrationBaseTest;
+
+import javax.jms.JMSException;
+import javax.management.*;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import javax.naming.NamingException;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.MalformedURLException;
+import java.rmi.RemoteException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * These test will validate metrics functionality in message broker.
+ * This class has 4 test cases.
+ * 1. Metrics reporting functionality test case
+ * 2. Validate queue subscribers count metric
+ * 3. Validate topic subscribers count metric
+ */
+public class MetricsTestCase extends MBIntegrationBaseTest {
+
+	private String sessionCookie;
+
+	/**
+	 * Initializing test case
+	 *
+	 * @throws XPathExpressionException
+	 */
+	@BeforeClass(alwaysRun = true) public void init()
+			throws XPathExpressionException, MalformedURLException, AutomationUtilException,
+			       MetricsConfigException, MetricsLevelConfigException, RemoteException,
+			       LoginAuthenticationExceptionException {
+		super.init(TestUserMode.SUPER_TENANT_USER);
+
+		LoginLogoutClient loginLogoutClientForAdmin =
+				new LoginLogoutClient(super.automationContext);
+		sessionCookie = loginLogoutClientForAdmin.login("admin", "admin", "localhost");
+	}
+
+
+	/**
+	 * This test case will validate queue subscriber count metric.
+	 * Steps -
+	 * 1. Create 5 subscribers and start sending messages.
+	 * 2. Report metric.
+	 * 3. Validate metric.
+	 * @throws AndesClientConfigurationException
+	 * @throws JMSException
+	 * @throws NamingException
+	 * @throws IOException
+	 * @throws AndesClientException
+	 */
+	@Test(groups = "wso2.mb", description = "Queue subscribers count metric test case")
+	public void performQueueSubscriberCountTestCase()
+			throws AndesClientConfigurationException, JMSException, NamingException, IOException,
+			       AndesClientException, InterruptedException, AutomationUtilException,
```

Comment:
Thread needs to sleep until publishers and clients start. If thread sends metrics data without sleeping, we might get incorrect values.